examples/echoserver/echoserver.c

```diff
@@ -2030,6 +2030,42 @@ static int wsUserAuth(byte authType,
 }
 
 
+#ifdef WOLFSSH_SFTP
+static int SetDefaultSftpPath(WOLFSSH* ssh, const char* defaultSftpPath)
+{
+    char path[WOLFSSH_MAX_FILENAME];
+    char realPath[WOLFSSH_MAX_FILENAME];
+    int ret = 0;
+
+    if (defaultSftpPath == NULL) {
+    #ifdef USE_WINDOWS_API
+        if (GetCurrentDirectoryA(sizeof(path), path) == 0) {
+            ret = -1;
+        }
+    #else
+        if (getcwd(path, sizeof(path)) == NULL) {
+            ret = -1;
+        }
+    #endif
+    }
+    else {
+        WSTRNCPY(path, defaultSftpPath, sizeof(path));
```

Comment:
Yeah, we should probably check that. At this point we know we aren't going to have a good path, that it will be truncated. We don't need to try to use it in a function to check it later. Also, the truncated path might point to some other file.